spec/services/preserved_object_handler_confirm_version_spec.rb

```diff
@@ -222,59 +201,50 @@
+          allow(cm).to receive(:save!).and_raise(ActiveRecord::ActiveRecordError, 'foo')
+        end
 
-          it 'DB_UPDATE_FAILED error' do
-            expect(results).to include(a_hash_including(AuditResults::DB_UPDATE_FAILED))
-          end
+        it 'DB_UPDATE_FAILED error' do
+          expect(po_handler.confirm_version).to include(a_hash_including(AuditResults::DB_UPDATE_FAILED))
         end
       end
 
-      it 'calls CompleteMoab.save! (but not PreservedObject.save!) if the existing record is altered' do
-        po = create :preserved_object
-        allow(PreservedObject).to receive(:find_by).with(druid: druid).and_return(po)
-        cm = create :complete_moab, preserved_object: po
-        allow(CompleteMoab).to receive(:find_by).with(preserved_object: po, moab_storage_root: ms_root).and_return(cm)
-        allow(po_handler).to receive(:moab_validation_errors).and_return([])
+      describe 'calls CompleteMoab.save! (but not PreservedObject.save!)' do
+        before do
+          allow(PreservedObject).to receive(:find_by).with(druid: druid).and_return(po)
+          allow(po.complete_moabs).to receive(:find_by!).with(moab_storage_root: ms_root).and_return(cm)
+        end
 
-        allow(po).to receive(:save!)
-        allow(cm).to receive(:save!)
-        po_handler.confirm_version
-        expect(po).not_to have_received(:save!)
-        expect(cm).to have_received(:save!)
-      end
-      it 'calls CompleteMoab.save! (but not PreservedObject.save!) if the existing record is NOT altered' do
-        po_handler = described_class.new(druid, 1, 1, ms_root)
-        po = create :preserved_object
-        allow(PreservedObject).to receive(:find_by).with(druid: druid).and_return(po)
-        cm = create :complete_moab, preserved_object: po
-        allow(CompleteMoab).to receive(:find_by).with(preserved_object: po, moab_storage_root: ms_root).and_return(cm)
-
-        allow(po).to receive(:save!)
-        allow(cm).to receive(:save!)
-        po_handler.confirm_version
-        expect(cm).to have_received(:save!)
-        expect(po).not_to have_received(:save!)
+        it 'if the existing record is altered' do
+          allow(po_handler).to receive(:moab_validation_errors).and_return([])
+          expect(cm).to receive(:save!)
+          expect(po).not_to receive(:save!)
+          po_handler.confirm_version
+        end
+        it 'calls CompleteMoab.save! (but not PreservedObject.save!) if the existing record is NOT altered' do
+          po_handler = described_class.new(druid, 1, 1, ms_root)
+          allow(po_handler).to receive(:moab_validation_errors).and_return([]) # different po_handler now
+          expect(cm).to receive(:save!)
+          expect(po).not_to receive(:save!)
+          po_handler.confirm_version
+        end
       end
+
       it 'logs a debug message' do
-        msg = "confirm_version #{druid} called"
         allow(Rails.logger).to receive(:debug)
```

Comment:
total nitpick, but this line is probably superfluous now, right?

Reply:
No, the point is it receives multiple calls to debug, but we only care about confirming one.  If you only set the expectation, you would get an error like `expected :x with 'val', but got :x with 'other'`